### PR TITLE
feat: store software upgrade plan and refresh data at upgrade height

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description

Closes: #XXXX

jira: https://forbole.atlassian.net/browse/BDU-604

Background: 
Sifchain hard-coded `validator minimum commission` in a new release and updated it with `software upgrade`. Without any `MsgEditValidator` involved, the `validator commission %` could not be updated and were then incorrect. 

This PR stores `software upgrade proposal's upgrade plan` and uses the `upgrade` module to check if there's an upgrade plan at each height, if yes it will proceed with neccessary updates of data. Currently it's only updating the validator infos (the rest of the data I believe we have `periodic task` or `handle block` which should be enough). We can add more  in the future if similar cases pop up. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: modules/upgrade/module.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+	"github.com/forbole/juno/v3/modules"
+)
+
+var (
+	_ modules.Module      = &Module{}
+	_ modules.BlockModule = &Module{}
+)
+
+// Module implements the consensus utils
```
Comment (MonikaCat) [inline modules/upgrade/module.go]:
```suggestion
// Module represents the x/upgrade module
```

**Review by MonikaCat — CHANGES_REQUESTED** [review]
@huichiaotsou Can you also add a handler to remove the software upgrade details from `software_upgrade_plan` table after the upgrade height has been passed?

**Review by MonikaCat — APPROVED** [review]
tested, LGTM

**Review by rmontagnin — APPROVED** [review]
LGTM, great job @huichiaotsou! :clap:

**Pull request merged**